mail_notification_custom_subject/models/mail_message_custom_subject.py

```diff
@@ -0,0 +1,243 @@
+# Copyright 2020 Tecnativa - João Marques
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import babel
+import copy
+import datetime
+import dateutil.relativedelta as relativedelta
+import logging
+
+import functools
+from werkzeug import urls
+
+from odoo import _, api, fields, models, tools
+from odoo.exceptions import UserError
+from odoo.tools import pycompat
+
+_logger = logging.getLogger(__name__)
+
+
+def format_date(env, date, pattern=False):
```

Comment:
Why redefining all these methods? Can't you import them?
This is linked with the following comment